component/metrics/remotewrite/types.go

```diff
@@ -0,0 +1,132 @@
+package remotewrite
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/prometheus/prometheus/config"
+
+	"github.com/grafana/agent/component/metrics"
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+	"github.com/grafana/agent/pkg/river"
+	common "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+)
+
+var (
+	// DefaultQueueConfig matches the default defined in prometheus
+	DefaultQueueConfig = QueueConfig{
```

Comment:
I thought about tying it directly, but decided we may want to shift or at least not be as tightly bound.